modules/reactor/src/meshgenerators/PinMeshGenerator.C

```diff
@@ -113,17 +120,42 @@ PinMeshGenerator::PinMeshGenerator(const InputParameters & parameters)
   _mesh_dimensions = getReactorParam<int>("mesh_dimensions");
   _mesh_geometry = getReactorParam<std::string>("mesh_geometry");
 
+  if (_is_assembly)
+  {
+    auto assembly_pitch = getReactorParam<Real>("assembly_pitch");
+    if (assembly_pitch != _pitch)
+      mooseError("Pitch defined in PinMeshGenerator must match assembly_pitch defined in "
+                 "ReactorMeshParams if use_as_assembly is set to true");
+  }
+
   if (_extrude && _mesh_dimensions != 3)
     mooseError("This is a 2 dimensional mesh, you cannot extrude it. Check your ReactorMeshParams "
                "inputs\n");
   if (_extrude && (!hasReactorParam("top_boundary_id") || !hasReactorParam("bottom_boundary_id")))
     mooseError("Both top_boundary_id and bottom_boundary_id must be provided in ReactorMeshParams "
                "if using extruded geometry");
 
-  if (_intervals.size() != (_ring_radii.size() + _duct_halfpitch.size() + 1))
-    mooseError("The number of mesh intervals must be equal to the number of annular regions + the "
-               "number of duct regions + 1"
-               " for the region between the rings and ducts\n");
+  if (_homogenized)
+  {
+    if (_mesh_geometry == "Square")
+      mooseError("Homogenization in PinMeshGenerator is only supported for hexagonal geometries");
+    const std::vector<std::string> disallowed_parameters = {
+        "num_sectors", "ring_radii", "duct_halfpitch", "mesh_intervals"};
+    for (const auto & parameter : disallowed_parameters)
+      if (isParamValid(parameter))
```

Comment:
if you use isParamSetByUser instead of isParamValid you can leave a default to mesh_intervals